pages/reviews/[id].tsx

```diff
@@ -0,0 +1,97 @@
+import HeroCards from "@/components/hero-cards";
+import MusicPlaylist from "@/components/music-component";
+import Wrapper from "@/components/navbar-wrapper";
+import NewsLetterInput from "@/components/news-letter-input";
+import { reviewEachPageDetails, heroCards } from "@/utils/mockdata";
+import { GetStaticProps } from "next";
+import Image, { StaticImageData } from "next/image";
+import React from "react";
+
+export const getStaticPaths = async () => {
+  //some asynchronous operation to a backend service maybe if available to get the amount of review on the data
+  const data = ["1", "2", "3", "4"];
+
+  // map data to an array of path objects with params (id)
+  const paths = data.map((path) => {
+    return {
+      params: { id: path }
+    };
+  });
+
+  return {
+    paths,
+    fallback: false
+  };
+};
+export const getStaticProps: GetStaticProps = async (context) => {
```

Comment:
we plan to use server side rendering not static generation, so we'll be using `getServerSideProps`, also not sure how the page routes will be, but for now lets leave all blog paths as `/blog/[id]`